### ISSUE TITLE
precalculate / reduce expressions with +/-

### ISSUE THREAD

**Issue by nihilus** [yegord/snowman#49]
An expression ending up with some thing like "x = y + 64 - 64 + 4;" should be reduced to "x = y + 4;"

**Comment by yegord** [comment]
Yes. Unfortunately, we get these expressions only after we have generated LikeC code. And LikeC is a bad representation for doing constant folding on it. So, I do not have good ideas on how to fix this.

**Comment by nihilus** [comment]
Got this as well:
```
    gp = 0x50000;
    gp = gp - 0x7c14;
```
which should be substituted into one statement.
NB that $gp is a 'global' register and is treated as such.
